x/wasm/internal/keeper/ioutil.go

```diff
@@ -5,20 +5,22 @@ import (
 	"compress/gzip"
 	"io"
 	"io/ioutil"
+
+	"github.com/CosmWasm/wasmd/x/wasm/internal/types"
 )
 
 // magic bytes to identify gzip.
 // See https://www.ietf.org/rfc/rfc1952.txt
 // and https://github.com/golang/go/blob/master/src/net/http/sniff.go#L186
 var gzipIdent = []byte("\x1F\x8B\x08")
 
-// limit max bytes read to prevent gzip bombs
-const maxSize = 400 * 1024
-
 // uncompress returns gzip uncompressed content or given src when not gzip.
-func uncompress(src []byte) ([]byte, error) {
-	if len(src) < 3 {
+func uncompress(src []byte, limit uint64) ([]byte, error) {
+	switch n := uint64(len(src)); {
+	case n < 3:
 		return src, nil
+	case n > limit:
+		return nil, types.ErrLimit
```

Comment:
This check also rejects compressed data > limit, which is smaller than limit when uncompressed. Is this intended?

Reply:
good edge case!
This can only happen when gzip does not compress the data but adds it header on top. I guess that is very unlikely to happen and the workaround would be to send uncompressed data.

Reply:
Right. If found the code a bit confusing to review, because it is not very clear which parts deals with compressed and which with uncompressed data. Some checks operate on both.